dsp/transform/hilbert_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"fmt"
+	"testing"
+
+	"gonum.org/v1/gonum/cmplxs"
+)
+
+func TestHilbertAnalytic(t *testing.T) {
+	const tol = 1e-10
+	testCases := []struct {
+		in   []float64
+		want []complex128
+	}{
+		{[]float64{0, 0, 0, 0}, []complex128{0, 0, 0, 0}},
```

Comment:
Please add a test fit a nil slice, and an empty slice

This should be covered

Reply:
These tests now pass, but I will note that these cases panic for the `CmplxFFT` methods that this transform is based on